components/script/dom/texttracklist.rs

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::TextTrackListBinding::{
+    self, TextTrackListMethods,
+};
+use crate::dom::bindings::reflector::reflect_dom_object;
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::bindings::str::DOMString;
+use crate::dom::eventtarget::EventTarget;
+use crate::dom::texttrack::TextTrack;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+
+#[dom_struct]
+pub struct TextTrackList {
+    eventtarget: EventTarget,
+    dom_tracks: DomRefCell<Vec<Dom<TextTrack>>>,
+}
+
+impl TextTrackList {
+    pub fn new_inherited(tracks: &[&TextTrack]) -> TextTrackList {
+        TextTrackList {
+            eventtarget: EventTarget::new_inherited(),
+            dom_tracks: DomRefCell::new(tracks.iter().map(|g| Dom::from_ref(&**g)).collect()),
+        }
+    }
+
+    pub fn new(window: &Window, tracks: &[&TextTrack]) -> DomRoot<TextTrackList> {
+        reflect_dom_object(
+            Box::new(TextTrackList::new_inherited(tracks)),
+            window,
+            TextTrackListBinding::Wrap,
+        )
+    }
+
+    pub fn item(&self, idx: u32) -> Option<DomRoot<TextTrack>> {
+        self.dom_tracks
+            .borrow()
+            .get(idx as usize)
+            .map(|t| DomRoot::from_ref(&**t))
+    }
+
+    pub fn find(&self, track: &TextTrack) -> Option<usize> {
+        self.dom_tracks
+            .borrow()
+            .iter()
+            .enumerate()
+            .filter(|(_, t)| t.id() == track.id())
```

Comment:
Again, not correct.

Reply:
Updated to use `==` which after looking at gecko I think is the right thing to do.